marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/endpoints/collectorLib.sjs

```diff
@@ -19,13 +19,23 @@ const DataHub = require("/data-hub/5/datahub.sjs");
 
 class CollectorLib {
 
+  /**
+   * Determine the sourceQuery from the given options and stepDefinition and then prepare it for evaluation by the
+   * collector endpoint.
+   *
+   * @param combinedOptions
+   * @param stepDefinition
+   * @return {string|*}
+   */
   prepareSourceQuery(combinedOptions, stepDefinition) {
     let sourceQuery = combinedOptions.sourceQuery;
 
     let isMergingStep = stepDefinition.name === 'default-merging' && stepDefinition.type === 'merging';
 
     if (isMergingStep) {
-      sourceQuery = fn.normalizeSpace(`cts.values(cts.pathReference('/matchSummary/URIsToProcess', ['type=string','collation=http://marklogic.com/collation/']), null, null, ${sourceQuery})`);
+      return fn.normalizeSpace(`cts.values(cts.pathReference('/matchSummary/URIsToProcess', ['type=string','collation=http://marklogic.com/collation/']), null, null, ${sourceQuery})`);
+    } else if (combinedOptions.sourceQueryIsScript) {
```

Comment:
if return from "if" code block, you do not need to use "else if", only "if" is enough.

Reply:
Good idea, I'll make the change.

Reply:
Updated - definitely reads nicer now.